patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -693,15 +691,15 @@
 +
 +        public ToolMaterial setRepairItem(ItemStack stack)
 +        {
-+            if (this.repairMaterial != null) throw new RuntimeException("Repair material has already been set");
++            if (!this.repairMaterial.func_190926_b()) throw new RuntimeException("Repair material has already been set");
 +            if (this == WOOD || this == STONE || this == GOLD || this == IRON || this == DIAMOND) throw new RuntimeException("Can not change vanilla tool repair materials");
 +            this.repairMaterial = stack;
 +            return this;
 +        }
 +
 +        public ItemStack getRepairItemStack()
 +        {
-+            if (repairMaterial != null) return repairMaterial;
++            if (!repairMaterial.func_190926_b()) return repairMaterial;
```

Comment:
We should return `repairMaterial.copy()` to make sure that the repair material doesn't get made invalid accidentally.